src/Main.java

```diff
@@ -1,5 +1,154 @@
+        public String decode(String input, int key) {
+            StringBuilder output = new StringBuilder();
+            for (int i = 0; i < input.length(); i++) {
+                char c = input.charAt(i);
+                if (Character.isLetter(c)) {
+                    c = (char) (input.charAt(i) - key);
+                    if ((Character.isLowerCase(input.charAt(i)) && c < 'a')
+                            || (Character.isUpperCase(input.charAt(i)) && c < 'A')) {
+                        c = (char) (input.charAt(i) + (26 - key));
+                    }
+                }
+                output.append(c);
+            }
+            return output.toString();
+        }
+    }
+
+    class UnicodeCipher implements IEncoder {
+        @Override
+        public String encode(String input, int key) {
+            StringBuilder result = new StringBuilder();
+            for (char c : input.toCharArray()) {
+                result.append((char) (c + key));
+            }
+            return result.toString();
+        }
+
+        @Override
+        public String decode(String input, int key) {
+            StringBuilder result = new StringBuilder();
+            for (char c : input.toCharArray()) {
+                result.append((char) (c - key));
+            }
+            return result.toString();
+        }
+    }
+
+    class EncoderFactory {
+        public IEncoder create(String alg) {
+            alg = alg.toUpperCase();
+
+            return switch (alg) {
+                case "SHIFT" -> new ShiftCipher();
+                case "UNICODE" -> new UnicodeCipher();
+                default -> throw new IllegalStateException("Unexpected value: " + alg);
+            };
+        }
+    }
+
+    class Solution {
+        private String input = "";
+        private String algorithm = "";
+        private String encode;
+        private String data = "";
+        private int key = 0;
+        private String inFile = "";
+        private String inFileData = "";
+        private String outFile = "";
+        private String output;
+
+        public void run(String[] args) {
+            argumentsParse(args);
+            inputInFile();
+            encodeAndOutput();
+        }
+
+        private void argumentsParse(String[] args) {
+            for (int i = 0; i < args.length; i++) {
+                switch (args[i]) {
+                    case "-mode" -> encode = args[i + 1];
+                    case "-data" -> data = args[i + 1];
+                    case "-alg" -> algorithm = args[i + 1];
+                    case "-key" -> key = Integer.parseInt(args[i + 1]);
+                    case "-in" -> inFile = args[i + 1];
+                    case "-out" -> outFile = args[i + 1];
+                }
+            }
+        }
+
+        private void inputInFile() {
+
+            if (!this.inFile.isBlank()) {
+                try (Scanner scanner = new Scanner(new File(this.inFile))) {
+                    this.inFileData = scanner.nextLine();
```

Comment:
scanner.nextLine is reading only actual line, if file includes multiple lines only first one will be encoded or decoded.